### PR TITLE
feat: adding a delete_event

### PR DESCRIPTION
**Pull request by claire1618**
This code enables the delete webhook in Vela, so users can select to trigger a build when a branch or tag are deleted.

### CODE REVIEW

File: database/repo.go [hunk 1]
```diff
@@ -66,6 +66,7 @@ type Repo struct {
 	AllowDeploy  sql.NullBool   `sql:"allow_deploy"`
 	AllowTag     sql.NullBool   `sql:"allow_tag"`
 	AllowComment sql.NullBool   `sql:"allow_comment"`
+	AllowDelete  sql.NullBool   `sql:"allow_delete"`
```
Comment (ecrupper) [inline database/repo.go:69]: No need for this field with the `AllowEvents` struct.

File: item_test.go [hunk 2]
```diff
@@ -59,6 +59,7 @@ func TestTypes_ToItem(t *testing.T) {
 		AllowPush:   &booL,
 		AllowDeploy: &booL,
 		AllowTag:    &booL,
+		AllowDelete: &booL,
```
Comment (ecrupper) [inline item_test.go:62]: Shouldn't need this field for library.Repo

File: library/repo.go [hunk 3]
```diff
@@ -47,6 +48,7 @@ func (r *Repo) Environment() map[string]string {
 	return map[string]string{
 		"VELA_REPO_ACTIVE":        ToString(r.GetActive()),
 		"VELA_REPO_ALLOW_COMMENT": ToString(r.GetAllowComment()),
+		"VELA_REPO_ALLOW_DELETE":  ToString(r.GetAllowDelete()),
```
Comment (ecrupper) [inline library/repo.go:51]: Won't need these environment values

File: library/repo.go [hunk 4]
```diff
@@ -35,6 +35,7 @@ type Repo struct {
 	AllowDeploy  *bool     `json:"allow_deploy,omitempty"`
 	AllowTag     *bool     `json:"allow_tag,omitempty"`
 	AllowComment *bool     `json:"allow_comment,omitempty"`
+	AllowDelete  *bool     `json:"allow_delete,omitempty"`
```
Comment (ecrupper) [inline library/repo.go:38]: `AllowEvents` struct should have this info

File: library/repo.go [hunk 5]
```diff
@@ -765,6 +794,8 @@ func (r *Repo) SetApproveBuild(v string) {
 }
 
 // EventAllowed determines whether or not an event is allowed based on the repository settings.
+//
+//nolint:nakedret // not necessary to watch for here
```
Comment (ecrupper) [inline library/repo.go:798]: Let's follow the linter here and no longer do a naked return for this

File: library/secret.go [hunk 6]
```diff
@@ -79,6 +79,8 @@ func (s *Secret) Match(from *pipeline.Container) bool {
 		eACL = checkEvent(events, constants.EventDeploy)
 	case constants.EventComment:
 		eACL = checkEvent(events, constants.EventComment)
+	case constants.EventDelete:
```
Comment (ecrupper) [inline library/secret.go:82]: I think we should implement secrets allowed events the same way we do repo allowed events

**Review by ecrupper — CHANGES_REQUESTED** [review]
There shouldn't be a new field in `library.Repo` or `database.Repo` — only in `library.Events` should there be a new action set for delete events.

File: library/secret.go (hunk 6 above)
Comment (claire1618) [reply library/secret.go]: Can do, let me know if the changes I made are what you imagined

**Review by wass3rw3rk — COMMENTED** [review]
lgtm, do we want to add `delete` here as well: https://github.com/go-vela/types/blob/26e54c869418eb841c5cc1d31b3364459378c9b6/yaml/ruleset.go#L147-L163, so when you specify just `delete` it will add both actions for you? not saying we have to, just asking to get thoughts.

**Review by KellyMerrick — APPROVED** [review]
lgtm

**Review by ecrupper — APPROVED** [review]

**Pull request merged**